frontends/main/src/app-pages/DashboardPage/CoursewareDisplay/DashboardCard.tsx

```diff
@@ -266,7 +262,12 @@ const DashboardCard: React.FC<DashboardCardProps> = ({
     />
   )
   const desktopLayout = (
-    <CardRoot data-testid="enrollment-card-desktop">
+    <CardRoot
+      screenSize="desktop"
+      data-testid="enrollment-card-desktop"
+      as={Component}
+      className={className}
+    >
```

Comment:
@gumaerc I moved the desktop-only / mobile-only behavior to CardRoot.

My motivation was:
- I want `Component` to be the root component (So that if we nest cards in a list, we can specify `Component="li"`
- Classname should similarly apply to the root component.

Alternatively, could have moved classname, testid, and Component onto `DesktopOnly` / `MobileOnly`. Not sure what's more preferable.